code/modules/recycling/disposal-unit.dm

```diff
@@ -527,6 +527,177 @@
+				if(auto_close_on_mob)
+					spawn(auto_close_on_mob)
+						close()
+		for(var/obj/item/O in loc)
+			if(!O.throwing || !O.anchored)
+				O.loc = src
+	if(auto_close)
+		spawn(auto_close)
+			close()
+	return 1
+
+
+/obj/machinery/disposal/trapdoor/proc/close()
+	if(open)
+		flick("closing", src)
+		icon_state = "closed"
+		playsound(src.loc, sound_open, 100, 1)
+		open = FALSE
+		return 1
+
+/obj/machinery/disposal/trapdoor/New(loc,var/obj/structure/disposalconstruct/make_from)
+	..()
+	stored.ptype = DISP_END_CHUTE
+	spawn(5)
+		trunk = locate() in loc
+		if(trunk)
+			trunk.linked = src
+
+/obj/machinery/disposal/trapdoor/Crossed(AM as mob|obj)
+	if(open)
+		if(istype(AM, /mob/living/))
+			var/mob/living/M = AM
+			if(M.floating)
+				return
+			M.loc = src
+			trap_flush()
+			if(auto_close_on_mob)
+				spawn(auto_close_on_mob)
+					close()
+			return
+		if(istype(AM, /obj/item))
+			var/obj/item/O = AM
+			spawn(5)
+				if(O.throwing || O.anchored)
+					return
+				else if(O.loc == src.loc)
+					O.loc = src
+					trap_flush()
+
+/obj/machinery/disposal/trapdoor/MouseDrop_T(mob/living/target, mob/living/user)
+	if (!open)
+		return
+	if(istype(target))
+		push_mob_in(target, user)
+		return 1
+
+/obj/machinery/disposal/trapdoor/proc/push_mob_in(mob/living/target, mob/living/user)
+	if(target.buckled)
+		return
+	add_fingerprint(user)
+	if(user == target)
+		if(target.floating)
+			user.visible_message("[user] is attempting to dive into [src].", \
+				"<span class='notice'>You start diving into [src]...</span>")
+			if(do_mob(target, user, 10))
+				if (!loc)
+					return
+				target.loc = src
+				user.visible_message("[user] dives into [src].", \
+					"<span class='notice'>You dive into [src].</span>")
+				sleep(5)
+				trap_flush()
+				if(auto_close_on_mob)
+					spawn(auto_close_on_mob)
+						close()
+		else
+			user.visible_message("[user] is attempting to step on the edge of [src].", \
+				"<span class='notice'>You start attempting to step on the edge of [src]...</span>")
+			if(do_mob(target, user, 30))
+				return
+			var/chance = 25 // normal chance, 25% to fall inside
+			var/turf/T = get_turf(src) // to check if turf's wet | T: variable defined but not used
+			var/M = "fall inside"
+			var/U = "falls inside"
+			if(user.disabilities & CLUMSY)
+				chance = 80
+				M = "accidentally do a backward flip, falling inside"
+				U = "accidentally does a backward flip, falling inside"
+			else if(user.getBrainLoss() >= 50)
+				chance = 70
+				M = "close your eyes and boldly step forward"
+				U = "closes his eyes and boldly steps forward"
+			else if(T.wet && isobj(user.shoes) && user.shoes.flags&NOSLIP) //user.shoes: undefined var | user.shoes.flags: undefined var | T.wet: undefined var
+				chance = 60
+				M = "slip and fall inside"
+				U = "slips and falls inside"
+			if(prob(chance))
+				user.visible_message("[U] \the [src]!", "You [M] \the [src]!")
+				user.loc = loc
+				user.Stun(10)
+			target.loc = src.loc
```

Comment:
after user.stun  and before target.loc = src.loc put an "else" and fix the indentation